File: app/src/main/java/grey/smarthouse/ui/relaySettingsScreen/RelaySettingsActivity.java
package grey.smarthouse.ui.relaySettingsScreen;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import java.util.UUID;

import grey.smarthouse.ui.SingleFragmentActivity;

public class RelaySettingsActivity extends SingleFragmentActivity {

    private static final String EXTRA_RELAY_ID = "grey.smarthouse.relay_id";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addFragment();
    }

    @Override
    protected Fragment createFragment() {
        UUID relayId = (UUID) getIntent().getSerializableExtra(EXTRA_RELAY_ID);
        return RelaySettingsFragment.newInstance(relayId);
    }

    public static Intent NewIntent(Context packageContext, UUID relayId) {
        Intent intent = new Intent(packageContext, RelaySettingsActivity.class);
        intent.putExtra(EXTRA_RELAY_ID, relayId);
        return intent;
    }
}

File: app/src/main/java/grey/smarthouse/ui/relaySettingsScreen/RelaySettingsActivity.kt
package grey.smarthouse.ui.relaySettingsScreen

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import androidx.fragment.app.Fragment
import grey.smarthouse.R

class RelaySettingsActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_default)
        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(R.id.fragment_container)
        if (fragment == null) {
            fragment = createFragment()
            fragment?.let {
                fm.beginTransaction()
                    .add(R.id.fragment_container, it)
                    .commit()
            }
        }
    }

    fun createFragment(): Fragment? {
        val relayNum = intent.getSerializableExtra(R.string.EXTRA_RELAY_ID.toString()) as Int
        return RelaySettingsFragment.newInstance(relayNum)
    }

    companion object {
        fun NewIntent(packageContext: Context, num: Int): Intent {
            val intent = Intent(packageContext, RelaySettingsActivity::class.java)
            intent.putExtra(R.string.EXTRA_RELAY_ID.toString(), num)
            return intent
        }
    }
}
